File: src/lib/entities/PlayingCard.ts
import type PlayingCardSuite from "$lib/enums/PlayingCardSuit"
import type PlayingCardValue from "$lib/enums/PlayingCardValue"

export default class PlayingCard {
	public suit: PlayingCardSuite
	public value: PlayingCardValue

	constructor(suit: PlayingCardSuite, value: PlayingCardValue) {
		this.suit = suit
		this.value = value
	}

	toString(): string {
		return `${this.value} of ${this.suit}`
	}
}

File: src/entities/playing_card.rs
use std::fmt::{Display, Formatter};
use bevy::prelude::Component;
use crate::enums::{PlayingCardSuit, PlayingCardValue};
use crate::textures::PlayingCardTexture;

/// Playing card with value and suit
#[derive(Component, Copy, Clone)]
pub struct PlayingCard {
	pub(crate) value: PlayingCardValue,
	pub(crate) suit: PlayingCardSuit,
}

impl PlayingCard {
	pub fn new(value: PlayingCardValue, suit: PlayingCardSuit) -> Self {
		Self { value, suit }
	}

	pub fn sprite_index(&self) -> usize {
		let row = self.suit as usize;
		let column = self.value as usize;
		column + (row * PlayingCardTexture::atlas_columns())
	}
}

impl Display for PlayingCard {
	fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
		write!(f, "{} of {}", self.value, self.suit)
	}
}